Core/Entities/Role.cs

```diff
@@ -0,0 +1,9 @@
+﻿using Microsoft.AspNetCore.Identity;
+
+namespace Core.Entities
+{
+    public class Role : IdentityRole<Guid>
+    {
+        public ICollection<User> Users { get; set; }
+    }
```

Comment:
Role should not contain users, this feels weird. It's rather you should have a map where each user is assigned a role.